### PR TITLE
Clarify GitHub App permissions

### PR DESCRIPTION
**Pull request by kathy-t**
https://github.com/dockstore/dockstore/issues/4613

This PR clarifies who can install the GitHub App in an organization. From my testing, only organization admins and repository admins were able to.

A repository admin who is not an organization admin has a harder time installing the GitHub App, but I outlined the two scenarios that are possible.

Local screenshot:
image

### CODE REVIEW

File: docs/getting-started/github-apps/snippet--installation.rst [hunk 1]
```diff
@@ -33,10 +33,24 @@ You should now see the organization and the repositories you chose to keep track
 .. figure:: /assets/images/docs/my-services-filled.png
 
 
-A note on permissions
-~~~~~~~~~~~~~~~~~~~~~
+A note on permissions when installing the GitHub App to a GitHub organization
```
Comment (coverbeck) [inline docs/getting-started/github-apps/snippet--installation.rst:36]: I suggest "installing a GitHub App" or "installing the Dockstore GitHub App", with a ripple effect of how you refer to it in the next section, if you change it.

File: docs/getting-started/github-apps/snippet--installation.rst [hunk 2]
```diff
@@ -33,10 +33,24 @@ You should now see the organization and the repositories you chose to keep track
...
+
+1. An organization admin configured the GitHub App to be installed on all repositories.
+
+   * Attempts to install the GitHub App on any repository in the organization will result in an error from GitHub, regardless of whether or not the repository is one that the user has admin access to.
+   * If it is a repository that the user has admin access to, the user can :ref:`check the repository to see if the app is indeed installed <Check GitHub App installation on repository>`.
+
+2. An organization admin configured the GitHub App to be installed on select repositories or the GitHub App is not configured at all. 
+
+   * The user can install the GitHub App on repositories that they are an admin of.
+   * If they attempt to install the app on a repository that they are not an admin of, it may look like the app install successfully, but in reality, it did not install. Instead, an organization admin would receive a request to install the GitHub App on the repository.
```
Comment (coverbeck) [inline docs/getting-started/github-apps/snippet--installation.rst:51]: Suggest "would receive a request" -> "will receive a request"

**Review by coverbeck — COMMENTED** [review]
We just created a release branch yesterday (after your comment on my PR :) ). Can you please retarget this to the release branch.

**Comment by kathy-t** [comment]
> We just created a release branch yesterday (after your comment on my PR :) ). Can you please retarget this to the release branch.

Hmm, the milestone got changed from 1.13 to 1.13.x, but if the release branch is still around then should I change the ticket's milestone back to 1.13?

**Comment by coverbeck** [comment]
> Hmm, the milestone got changed from 1.13 to 1.13.x, but if I target this PR to the release branch then should I change the ticket's milestone back to 1.13?

Might as well; I don't see any reason to delay this for later hotfix -- it looks good.

**Comment by kathy-t** [comment]
Retargeted to 1.13 branch and changed the ticket milestone back to 1.13

**Review by coverbeck — APPROVED** [review]

**Review by aofarrel — CHANGES_REQUESTED** [review]
I do like that this is comprehensive, but since this snippet shows up on multiple pages and is a little too detailed for the places it gets embedded. 

I think its best to move everything starting with "There are two scenarios to be aware of when installing the Dockstore GitHub App as a repository admin who is not an organization admin..." to be moved to the FAQ page. ie, move everything from the second sentence of line 41 thru the end of line 51 to the FAQ.

**Review by svonworl — APPROVED** [review]

File: docs/getting-started/github-apps/github-apps-troubleshooting-tips.rst [hunk 3]
```diff
@@ -62,6 +64,18 @@ Perhaps. GitHub permissions can quickly get complicated, as it involves two leve
 
 First of all, you can only configure already-installed GitHub Apps for organizations you are not an admin in if you go through the app installation process again. Be aware that in this scenario, you can only add repos that you have admin access to, not just maintainer access.
 
+There are two scenarios to be aware of when installing the Dockstore GitHub App as a repository admin who is not an organization admin.
+
+1. An organization admin configured the Dockstore GitHub App to be installed on all repositories.
```
Comment (aofarrel) [inline docs/getting-started/github-apps/github-apps-troubleshooting-tips.rst:69]: Would reword this to "Another person, who is an organization admin,"

File: docs/getting-started/github-apps/github-apps-troubleshooting-tips.rst [hunk 4]
```diff
@@ -62,6 +64,18 @@ Perhaps. GitHub permissions can quickly get complicated, as it involves two leve
...
+
+   * Attempts to install the Dockstore GitHub App on any repository in the organization will result in an error from GitHub, regardless of whether or not the repository is one that the user has admin access to.
+   * If it is a repository that the user has admin access to, the user can :ref:`check the repository to see if the app is indeed installed <Check GitHub App installation on repository>`.
+
+2. An organization admin configured the Dockstore GitHub App to be installed on select repositories or the Dockstore GitHub App is not configured at all.
```
Comment (aofarrel) [inline docs/getting-started/github-apps/github-apps-troubleshooting-tips.rst:74]: Would reword this to "The Dockstore GitHub App is not configured at all, or another person who is an organization admin configured the Dockstore GitHub to be installed on only select repositories in the organization."

File: docs/getting-started/github-apps/github-apps-troubleshooting-tips.rst [hunk 5]
```diff
@@ -62,6 +64,18 @@ Perhaps. GitHub permissions can quickly get complicated, as it involves two leve
...
+   * If it is a repository that the user has admin access to, the user can :ref:`check the repository to see if the app is indeed installed <Check GitHub App installation on repository>`.
+
+2. An organization admin configured the Dockstore GitHub App to be installed on select repositories or the Dockstore GitHub App is not configured at all. 
+
+   * The user can install the Dockstore GitHub App on repositories that they are an admin of.
+   * If they attempt to install the app on a repository that they are not an admin of, it may look like the app install successfully, but in reality, it did not install. Instead, an organization admin will receive a request to install the Dockstore GitHub App on the repository. 
+
 You may also still run into scenarios where your changes appear to not get saved, even though GitHub will not throw an error. For example, if you tried to give the GitHub App access to databiosphere/analysis_pipeline_wdl, and upon re-installation into the DataBiosphere organization, you do not see databiosphere/analysis_pipeline_wdl in the list of repositories it already has access to, there is a good chance GitHub is blocking you.
```
Comment (aofarrel) [inline docs/getting-started/github-apps/github-apps-troubleshooting-tips.rst:79]: would change "and upon re-installation into the DataBiosphere organization," to "and upon re-installation of the app into the DataBiosphere organization,"

File: docs/getting-started/github-apps/github-apps-troubleshooting-tips.rst [hunk 6]
```diff
@@ -62,6 +64,18 @@ Perhaps. GitHub permissions can quickly get complicated, as it involves two leve
 
 First of all, you can only configure already-installed GitHub Apps for organizations you are not an admin in if you go through the app installation process again. Be aware that in this scenario, you can only add repos that you have admin access to, not just maintainer access.
 
+There are two scenarios to be aware of when installing the Dockstore GitHub App as a repository admin who is not an organization admin.
+
+1. An organization admin configured the Dockstore GitHub App to be installed on all repositories.
+
+   * Attempts to install the Dockstore GitHub App on any repository in the organization will result in an error from GitHub, regardless of whether or not the repository is one that the user has admin access to.
+   * If it is a repository that the user has admin access to, the user can :ref:`check the repository to see if the app is indeed installed <Check GitHub App installation on repository>`.
```
Comment (aofarrel) [inline docs/getting-started/github-apps/github-apps-troubleshooting-tips.rst:72]: This doesn't answer the question about whether all-repo access applies to new repos. What happens in this scenario?
1. Adam Organization-Admin installs app to have universal repo access (ie access to all repos)
2. Bob Repo-Admin later makes a new repo on the organization

Does Bob need to install the app at all or is the new repo already added?

File: docs/getting-started/github-apps/github-apps-troubleshooting-tips.rst (hunk 3 above)
Comment (aofarrel) [reply docs/getting-started/github-apps/github-apps-troubleshooting-tips.rst:69]: You may also want to reword "to be installed on all repositories" to something like "to have universal access to all repositories in the organization" because installing manually on all repos works differently, I think.

**Review by aofarrel — CHANGES_REQUESTED** [review]
Upon a second re-read, I think we can make this a little bit clearer (see comments). It is very difficult to break down GH's complicated permissions in an understandable way, so hats off to you for this.

File: docs/getting-started/github-apps/github-apps-troubleshooting-tips.rst (hunk 6 above)
Comment (denis-yuen) [reply docs/getting-started/github-apps/github-apps-troubleshooting-tips.rst:72]: Screenshot from 2022-09-16 10-10-37

This is more of a GitHub question than a Dockstore question, but GitHub does say that it would apply to future repos

File: docs/getting-started/github-apps/github-apps-troubleshooting-tips.rst [hunk 7]
```diff
@@ -62,6 +64,18 @@ Perhaps. GitHub permissions can quickly get complicated, as it involves two leve
...
+
+2. An organization admin configured the Dockstore GitHub App to be installed on select repositories or the Dockstore GitHub App is not configured at all. 
+
+   * The user can install the Dockstore GitHub App on repositories that they are an admin of.
+   * If they attempt to install the app on a repository that they are not an admin of, it may look like the app install successfully, but in reality, it did not install. Instead, an organization admin will receive a request to install the Dockstore GitHub App on the repository.
```
Comment (denis-yuen) [inline docs/getting-started/github-apps/github-apps-troubleshooting-tips.rst:77]: app install -> app installed

**Review by denis-yuen — APPROVED** [review]
build issue though

File: docs/getting-started/github-apps/github-apps-troubleshooting-tips.rst (hunk 3 above)
Comment (kathy-t) [reply docs/getting-started/github-apps/github-apps-troubleshooting-tips.rst]: > You may also want to reword "to be installed on all repositories" to something like "to have universal access to all repositories in the organization" because installing manually on all repos works differently, I think.

I reworded it to "to be installed on all current and future repositories" to match the phrasing in Denis' [screenshot](https://github.com/dockstore/dockstore-documentation/pull/225#discussion_r973069384)

**Review by aofarrel — APPROVED** [review]
Thanks! This is really great!
